src/storage/src/hummock/store/state_store.rs

```diff
@@ -63,6 +67,17 @@ pub struct HummockStorageCore {
     hummock_version_reader: HummockVersionReader,
 }
 
+impl Drop for HummockStorageCore {
+    fn drop(&mut self) {
+        // If sending fails, it means that event_handler and event_channel have been destroyed, no
+        // need to handle failure
+        let _ = self.event_sender.send(HummockEvent::DestroyReadVersion {
```

Comment:
If we use a instance id guard returned from event handler, we don't have to send the event here.